core/rawdb/database.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 The go-ethereum Authors
```

Comment:
Could you help me understand why this code was moved to `rawdb`? It seems like `database` was a good fit for this. IMHO `rawdb` provides low-level utility methods for writing `core` blockchain abstractions whereas `database` provides more low-level database abstractions and some starter implementations (e.g. the in-memory and LevelDB ones). This is especially true because the return types for these methods are from `database`.

Reply:
Creating the low level database needs to be able to inject the freezer. The freezer needs access to higher level types (blocks, etc), so it needs rawdb. Putting it into `database` introduces a dependency loop.